src/printf/printf.c

```diff
@@ -1055,8 +1056,9 @@ static int _vsnprintf(out_fct_type out, char* buffer, const size_t maxlen, const
             }
           }
           // string output
-          while ((*p != 0) && (!(flags & FLAGS_PRECISION) || precision--)) {
+          while ((*p != 0) && (!(flags & FLAGS_PRECISION) || precision)) {
```

Comment:
Is this exactly equivalent? 

What if there are no precision flags? In that case, with the new code, precision gets decremented while before it wasn't.

Reply:
That is true, but ```precision``` is only used to exit the while loop, so this does not impact any functionality.